docs/CppInteroperability/Roadmap.md

```diff
@@ -0,0 +1,264 @@
+
+```
+// StatefulObject has object identity and reference semantcs: 
+// it should be constructed with "create" and used via a pointer.
+struct StatefulObject {
+  StatefulObject(const StatefulObject&) = delete;
+  StatefulObject() = delete;
+
+  StatefulObject *create() { return new StatefulObject(); }
+};
+```
+
+This type has specific reference semantics (described in the comment). The copy constructor is deleted, preventing some mis-use, but the semantics are largely not communicated. The C++ programmer is responsible for reading the comment to know how to use the type correctly; there is no concrete attribute that specifies `StatefulObject` should be used as a reference type. 
+
+In Swift, reference types have their own language construct (`class`es). Swift clearly defines patterns for naming, generic programming, value categories, error handling, etc. and it is rare that a codebase deviates from the standard practice. Idiomatic, expressive Swift code uses these well-defined programming patterns, so, in an effort to allow Swift programmings who use C++ APIs to have a similar experience, the Swift compiler should map C++ APIs to one of these specific Swift programming patterns. To be able to map *any* C++ APIs to *some* specific Swift pattern, the Swift compiler must have some semantic information about the API. This semantic information may be provided through annotations or heuristics in the compiler. In the above example, the user should annotate `StatefulObject` as a reference type so that the Swift compiler can import it as a Swift `class`. 
+
+To be able to follow the “Swift API model,” that is to have expressive APIs with strong idioms, the following goals are nessisary:
+
+* C++ interoperability aims to provide a clear, well-defined mapping for APIs that can be imported, and to draw a clear boundary between these APIs and APIs that cannot be imported. A user should be able to read the C++ interoperability documentation and have a good idea of both how much of their C++ API will be able to be imported in Swift and what that API will look like once it is imported. 
+* It is a goal to preserve the semantics (including performance semantics) that an API was written with by using information provided by the user to understand an API's semantics and mapping those semantics to a Swift idiom that represents the same thing.
+* It is *not* a goal to import every C++ API into Swift. APIs that are imported by Swift should feel Swifty and idiomatic. There must be an easy way to add annotations so that APIs become importable, especially when updating APIs directly is not an option (because the source code is not available). 
```

Comment:
Maybe you can move this up to right after the discussion of `StatefulObject` and talk about the role of annotations more generally.  So you want to talk about how there are default rules and assumptions, and how things that don't match those will need to need to be imported differently or made unavailable by default, and then you can talk about how that can be customized with annotations.  That will lead very nicely into the paragraph about understanding how the API will look in Swift, and that in turn leads nicely into this final paragraph about existing uses through bridging.